include/seqan3/io/alignment_file/format_sam.hpp

```diff
@@ -135,6 +140,277 @@ class alignment_file_format_sam
+              id_type                                                 & id,
+              offset_type                                             & offset,
+              ref_seq_type                                            & SEQAN3_DOXYGEN_ONLY(ref_seq),
+              ref_id_type                                             & ref_id,
+              ref_offset_type                                         & ref_offset,
+              align_type                                              & align,
+              flag_type                                               & flag,
+              mapq_type                                               & mapq,
+              mate_type                                               & mate,
+              tag_dict_type                                           & tag_dict,
+              e_value_type                                            & SEQAN3_DOXYGEN_ONLY(e_value),
+              bit_score_type                                          & SEQAN3_DOXYGEN_ONLY(bit_score))
+    {
+        static_assert(detail::decays_to_ignore_v<ref_offset_type> ||
+                      detail::is_type_specialisation_of_v<ref_offset_type, std::optional>);
+
+        using stream_buf_t = std::istreambuf_iterator<typename stream_type::char_type>;
+        auto stream_view = view::subrange<decltype(stream_buf_t{stream}), decltype(stream_buf_t{})>
+                               {stream_buf_t{stream}, stream_buf_t{}};
+
+        auto next_field = view::take_until_or_throw(is_char<'\t'>);
+
+        // these variables need to be stored to compute the ALIGNMENT
+        std::conditional_t<!detail::decays_to_ignore_v<ref_offset_type>,
+                           ref_offset_type,
+                           std::optional<size_t>> ref_offset_tmp{};
+        [[maybe_unused]] std::string ref_id_tmp{}; // in case reference information is given
+        [[maybe_unused]] std::conditional_t<!detail::decays_to_ignore_v<offset_type>, offset_type, size_t> offset_tmp{};
+        [[maybe_unused]] std::conditional_t<!detail::decays_to_ignore_v<offset_type>, offset_type, size_t> soft_clipping_end{};
+        [[maybe_unused]] std::vector<std::pair<char, size_t>> cigar{};
+        [[maybe_unused]] size_t ref_length{0}, seq_length{0}; // length of aligned part for ref and query
+
+        // Header
+        // -------------------------------------------------------------------------------------------------------------
+        if (is_char<'@'>(*std::ranges::begin(stream_view))) // we always read the header if present
+        {
+            assert(header_ptr != nullptr);
+            read_header(stream_view, header_ptr);
+        }
+
+        // Fields 1-5: ID FLAG REF_ID REF_OFFSET MAPQ
+        // -------------------------------------------------------------------------------------------------------------
+        read_field(stream_view | next_field, id);
+        std::ranges::next(std::ranges::begin(stream_view)); // skip tab
```

Comment:
again you are not protecting against multiple tabs. I think in SeqAn this was handled.
Since the fields always have a value by specification we do not jump over a field if it wasn't available unless the file is ill-formatted.

Reply:
I don't understand...
As you said, every field has a value. Two tabs behind each other would mean an ill-formatted file. So why do I need to protect against multiple tabs?

Reply:
So I looked into the SeqAn code and it seem we did not protect it there as well. So leave it as is. But in general when dealing with input that is possibly curated by hand I would expect any kind of wrong usage. Since I couldn't find an explicit violation of using multiple tab stops except "you shouldn't do it", I was simply assuming that we can just skip multiple tabs if they happen to be there. The file would be anyway ill-formatted if two tabs were intended.